javascript/modules/client/src/client-common/modules/web3.ts

```diff
@@ -32,6 +32,7 @@ export class Web3Module implements IClientWeb3Core {
     if (context.gasFeeEstimationFactor) {
       this._gasFeeEstimationFactor = context.gasFeeEstimationFactor;
     }
+    Object.freeze(Web3Module.prototype);
```

Comment:
Freezing the prototype is fine, but I think someone could still mutate the signer or the provider. 

We could declare a local `Map` that was not exported, in which each class could store its own private components `privSigners.set(this, signer)`, then access them like `privSigners.get(this)`.

The downside is that this would take references and never release them

Reply:
This applies to IPFS and GraphQL as well